sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCVCIssuancePlugin.java

```diff
@@ -135,13 +146,19 @@ public VCResult<JsonLDObject> getVerifiableCredentialWithLinkedDataProof(VCReque
             Template template=credentialTypeTemplates.get(supportedType);
             validateContextUrl(template,contextList);
         }
-        String osid = (identityDetails.containsKey("sub")) ? (String) identityDetails.get("sub") : null;
-        if (osid == null) {
-            log.error("Invalid request: osid is null");
+
+        String registrySearchField = (identityDetails.containsKey("sub")) ? (String) identityDetails.get("sub") : null;
+        if (registrySearchField == null) {
+            log.error("Invalid request: registrySearchField is null");
             throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
         }
-        String registryUrl=credentialTypeConfigMap.get(requestedCredentialType).get(REGISTRY_GET_URL);
-        Map<String,Object> responseRegistryMap =fetchRegistryObject(registryUrl+osid);
+        Map<String,Object> responseRegistryMap;
+        if(usePsut){
+            responseRegistryMap= fetchRegistryObjectByPSUT(registrySearchUrl,registrySearchField);
```

Comment:
Here registrySearchUrl should be credential specific, can we add one more config and take this ?

mosip.esignet.vciplugin.sunbird-rc.credential-type.InsuranceCredential.registry-search-url=http://10.3.148.107/registry/api/v1/Insurance/search

Reply:
done